assets/js/sections/qso.js

```diff
@@ -1,393 +1,1216 @@
+	$('#band_rx').val(favs[this.innerText].band_rx);
+	$('#band').val(favs[this.innerText].band);
+	$('#frequency_rx').val(favs[this.innerText].frequency_rx);
+	$('#frequency').val(favs[this.innerText].frequency);
+	$('#selectPropagation').val(favs[this.innerText].prop_mode);
+	$('#mode').val(favs[this.innerText].mode).change();
+});
+
+
+function del_fav(name) {
+	if (confirm("Are you sure to delete Fav?")) {
+		$.ajax({
+			url: base_url + 'index.php/user_options/del_fav',
+			method: 'POST',
+			dataType: 'json',
+			contentType: "application/json; charset=utf-8",
+			data: JSON.stringify({ "option_name": name }),
+			success: function (result) {
+				get_fav();
+			}
+		});
+	}
+}
+
+function get_fav() {
+	$.ajax({
+		url: base_url + 'index.php/user_options/get_fav',
+		method: 'GET',
+		dataType: 'json',
+		contentType: "application/json; charset=utf-8",
+		success: function (result) {
+			$("#fav_menu").empty();
+			for (const key in result) {
+				$("#fav_menu").append('<label class="dropdown-item" style="display: flex; justify-content: space-between;"><span id="fav_recall">' + key + '</span><span class="badge bg-danger" id="fav_del" name="' + key + '"><i class="fas fa-trash-alt"></i></span></label>');
 			}
-			$('#callsign').val("");
-			$("#callsign").focus();
+			favs = result;
 		}
 	});
+}
 
-	// Sanitize some input data
-	$('#callsign').on('input', function() {
-		$(this).val($(this).val().replace(/\s/g, ''));
-	});
-	$('#locator').on('input', function() {
-		$(this).val($(this).val().replace(/\s/g, ''));
+function save_fav() {
+	var payload = {};
+	payload.sat_name = $('#sat_name').val();
+	payload.sat_mode = $('#sat_mode').val();
+	payload.band_rx = $('#band_rx').val();
+	payload.band = $('#band').val();
+	payload.frequency_rx = $('#frequency_rx').val();
+	payload.frequency = $('#frequency').val();
+	payload.prop_mode = $('#selectPropagation').val();
+	payload.mode = $('#mode').val();
+	$.ajax({
+		url: base_url + 'index.php/user_options/add_edit_fav',
+		method: 'POST',
+		dataType: 'json',
+		contentType: "application/json; charset=utf-8",
+		data: JSON.stringify(payload),
+		success: function (result) {
+			get_fav();
+		}
 	});
+}
 
-	$('.callsign-suggest').hide();
 
-	setRst($(".mode").val());
+var bc_bandmap = new BroadcastChannel('qso_window');
+bc_bandmap.onmessage = function (ev) {
+	if (ev.data == 'ping' && qso_manual == 0) {
+		bc_bandmap.postMessage('pong');
+	}
+}
 
-	/* On Page Load */
-	var catcher = function() {
-		var changed = false;
-		$('form').each(function() {
-			if ($(this).data('initialForm') != $(this).serialize()) {
-				changed = true;
-				$(this).addClass('changed');
-			} else {
-				$(this).removeClass('changed');
+var bc = new BroadcastChannel('qso_wish');
+bc.onmessage = function (ev) {
+	if (qso_manual == 0) {
+		if (ev.data.ping) {
+			let message = {};
+			message.pong = true;
+			bc.postMessage(message);
+		} else {
+			// console.log(ev.data);
```

Comment:
@besynnerlig 

Sugestion 1: "Clear existing callsign and fill the new one"

```suggestion
			// console.log(ev.data);
			if ($("#callsign").val() != "") {
				reset_fields();
			}
```